src/pip/_internal/utils/temp_dir_kinds.py

```diff
@@ -0,0 +1,7 @@
+"""Identifiers for different temp directories used in pip.
+
+We use a common module with constant values to prevent typo-related bugs.
+"""
+
+build_env_kind = "build-env"
+req_build_kind = "req-build"
```

Comment:
Is there a particular reason to not use `CONSTANT_VALUE` names? pip seems to use that for globals.

Reply:
I was planning to file a follow-up PR that makes this similar to how this is handled in `pip._internal.configuration.kinds`. If @chrahunt wants to do that in this PR, he's welcome to. Otherwise, I'm happy to file a follow-up PR for addressing this using that mechanism. :)

https://github.com/pypa/pip/blob/f537db5021df46816969123ee1c565db514452ae/src/pip/_internal/configuration.py#L67

Reply:
Addressed [here](https://github.com/pypa/pip/pull/7677/files#diff-bd9cdc0022571f341d8a451e918e279bR24-R29) as suggested by @pradyunsg.